### PR TITLE
list of connected devices is displayed

### PR DESCRIPTION
**Pull request by zalun**
issue#33

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,6 +45,42 @@ app.get('/', function(req, res) {
   res.render('index');
 });
 
+function sendMachines(req, res, token, machines) {
```
Comment (marco-c) [inline index.js]: Nit: req is unused

File: index.js [hunk 2]
```diff
@@ -68,8 +104,7 @@ app.post('/register', function(req, res) {
       key: req.body.key,
       name: req.body.name,
       active: true
-    })
-    .then(function() {
+    }).then(function() {
```
Comment (marco-c) [inline index.js:103]: Nit: unneeded change

File: index.js [hunk 3]
```diff
@@ -45,6 +45,42 @@ app.get('/', function(req, res) {
...
+          machines[machineId] = machine;
+          resolve();
+        });
+    });
+  }
+  redis.smembers(token)
+    .then(function(ids) {
+      var promises = ids.map(machinePromise);
+      Promise.all(promises)
+        .then(() => sendMachines(req, res, token, machines));
```
Comment (marco-c) [inline index.js]: Can you inline `sendMachines` here and rename the function from `showMachines` to `sendMachines`?

File: test/test.js [hunk 4]
```diff
@@ -60,7 +60,8 @@ describe('mercurius', function() {
       })
       .expect(function(res) {
         assert.equal(res.status, 200);
-        assert.equal(res.text, tokenToUnregister);
+        assert.equal(res.body.token, tokenToUnregister);
+        assert(res.body.machines);
```
Comment (marco-c) [inline test/test.js]: Can you assert that this is an array?

**Comment by marco-c** [comment]
We should add a test for this (get list of machines after adding a machine).

**Comment by marco-c** [comment]
(Either adding it now or filing a followup to add it)

File: index.js (hunk 2 above)
Comment (zalun) [inline index.js:103]: Yes - it was just I realized we've got a convention of putting .then in the next line ...

File: test/test.js (hunk 4 above)
Comment (zalun) [inline test/test.js]: Do you mean object? Yes

Comment (zalun) [inline test/test.js]: I'm now wondering if it shouldn't actually be an array - I mean `machines` bit...
Will we need any direct access to this in the front-end?
All functions are gonna be created for each machine in a loop...

File: static/index.js [hunk 5]
```diff
@@ -103,22 +110,60 @@ function makeId(length) {
...
-      domToken.textContent = token;
-    } else {
-      showSection('registrationForm');
-    }
+  }).
+  then(function() {
+    localforage.getItem('token')
+    .then(function(token) {
+      if (token) {
+        showSection(null);
```
Comment (marco-c) [inline static/index.js]: Why this call?

File: test/test.js (hunk 4 above)
Comment (marco-c) [inline test/test.js]: Don't we need it to show the list of machines?

File: static/index.js (hunk 5 above)
Comment (zalun) [inline static/index.js]: good call - I don't remember writing it :D

File: index.js [hunk 6]
```diff
@@ -45,6 +45,41 @@ app.get('/', function(req, res) {
...
 });
 
+app.get('/devices/:token', function(req, res) {
+  return sendMachines(req, res, req.params.token);
+});
+
+// get machines for the token and send them along with the token
+function sendMachines(req, res, token) {
+  if (!token) {
+    throw('No token provided');
```
Comment (marco-c) [inline index.js]: `throw new Error('No token provided')` should improve the error message when this happens.

File: index.js [hunk 7]
```diff
@@ -45,6 +45,41 @@ app.get('/', function(req, res) {
...
+  }
+  var machines = {};
+  var machineId;
+  function machinePromise(machineId) {
+    return new Promise(function(resolve, reject) {
```
Comment (marco-c) [inline index.js]: No need to create a new Promise, you can simply do:

``` JavaScript
return redis.hgetall(machineId)
.then(function(machine) {
  machines[machineId] = machine;
});
```

File: index.js (hunk 2 above)
Comment (marco-c) [inline index.js:103]: Yes, the Q style (https://github.com/kriskowal/q)

**Pull request merged**